x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartTrainedModelDeploymentAction.java

```diff
@@ -270,6 +282,74 @@ private void deleteFailedDeployment(
 
     }
 
+    private void validateModelDefinition(TrainedModelConfig config, ActionListener<Void> listener) {
+        if (config.getLocation() instanceof IndexLocation == false) {
+            listener.onResponse(null);
+            return;
+        }
+        final String modelId = config.getModelId();
+        final String[] requiredSourceFields = new String[] {
+            TrainedModelDefinitionDoc.DEFINITION_LENGTH.getPreferredName(),
+            TrainedModelDefinitionDoc.TOTAL_DEFINITION_LENGTH.getPreferredName(),
+            TrainedModelDefinitionDoc.EOS.getPreferredName() };
+        String index = ((IndexLocation) config.getLocation()).getIndexName();
+        client.prepareSearch(index)
+            .setQuery(
+                QueryBuilders.constantScoreQuery(
+                    QueryBuilders.boolQuery()
+                        .filter(QueryBuilders.termQuery(TrainedModelConfig.MODEL_ID.getPreferredName(), modelId))
+                        .filter(
+                            QueryBuilders.termQuery(InferenceIndexConstants.DOC_TYPE.getPreferredName(), TrainedModelDefinitionDoc.NAME)
+                        )
+                )
+            )
+            .setFetchSource(requiredSourceFields, new String[0])
+            .setSize(MAX_NUM_DEFINITION_DOCS)
```

Comment:
What if there are > 100 definition docs? The summed sizes will never equal expected total.